sw/device/lib/testing/test_framework/check.h

```diff
@@ -238,13 +260,17 @@
  *
  * @param expr An expression which evaluates to a `status_t`.
  */
-#define CHECK_STATUS_OK(expr, ...)                 \
-  do {                                             \
-    status_t status_ = expr;                       \
-    if (!status_ok(status_)) {                     \
-      LOG_ERROR("CHECK-STATUS-fail: %r", status_); \
-      test_status_set(kTestStatusFailed);          \
-    }                                              \
+#define CHECK_STATUS_OK(expr, ...)                                          \
+  do {                                                                      \
+    status_t status_ = expr;                                                \
+    if (!status_ok(status_)) {                                              \
+      const char *msg;                                                      \
+      int32_t line;                                                         \
+      char mod_id[4] = {0};                                                 \
+      status_extract(status_, &msg, &line, mod_id);                         \
+      LOG_ERROR("CHECK-STATUS-failed: %s error @%s:%d", msg, mod_id, line); \
```

Comment:
Our `printf` implementation already has a custom specifier for `status_t`:
```
LOG_ERROR("CHECK-STATUS-failed: %r", status_);
```

Reply:
Hunnn, that's why it was `%r`. But it didn't work in a simulated test with Xcelium, so I think It's not implemented by the dv testbench.
In that case, I'll drop this commit and implement the `%r` in the testbench in a follow-up PR.